flink-python/src/main/java/org/apache/flink/python/util/PythonHelperScriptUtils.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.python.util;
+
+import org.apache.flink.util.OperatingSystem;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Nullable;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Utils used to install the python requirements via pip.
+ */
+public class PythonHelperScriptUtils {
+
+	private static final Logger LOG = LoggerFactory.getLogger(PythonHelperScriptUtils.class);
+
+	private static final int MAX_RETRY_TIMES = 3;
+
+	private static final String PYFLINK_UDF_RUNNER_SH = "pyflink-udf-runner.sh";
+	private static final String PYFLINK_UDF_RUNNER_BAT = "pyflink-udf-runner.bat";
+
+	private static final String GET_SITE_PACKAGES_PATH_SCRIPT =
+		"import sys;" +
+		"from distutils.dist import Distribution;" +
+		"install_obj = Distribution().get_command_obj('install', create=True);" +
+		"install_obj.prefix = sys.argv[1];" +
+		"install_obj.finalize_options();" +
+		"installed_dir = [install_obj.install_purelib];" +
+		"install_obj.install_purelib != install_obj.install_platlib and " +
+			"installed_dir.append(install_obj.install_platlib);" +
+		"print(installed_dir[0]);" +
+		"len(installed_dir) > 1 and " +
+			"print(installed_dir[1])";
+
+	private static final String CHECK_PIP_VERSION_SCRIPT =
+		"import sys;" +
+		"from pkg_resources import get_distribution, parse_version;" +
+		"pip_version = get_distribution('pip').version;" +
+		"print(parse_version(pip_version) >= parse_version(sys.argv[1]))";
+
+	private static final String GET_RUNNER_SCRIPT_DIR =
```

Comment:
The constant name should be consistent, e.g. if we add the suffix `_SCRIPT`, we should make sure that all the constants have this suffix.